### DISCUSSION TITLE
Tags become effectively invisible when changing locale...

### DISCUSSION THREAD

**Discussion by leeovery** [spatie/laravel-tags#294] [Ideas]
Hey,

**RE: Disable translatable tag functionality (or PR for workaround)?**

I have an app which is multi-lingual, but where we **do not** need translatable tags.

The problem arises where the user creates tags in english for example, and hence are saved under the `en` key. If they then change their language to German, those English tags are no longer returned without manually specifying the `en` key. The issue obviously compounds if the user then creates more tags in german. In our case, the tags are duplicated as if the `en` ones didn't exist (due to saving the whole model rather than patching).

I wonder if any thought has been given to perhaps disabling the translation abilities of the tags package? Id be happy to PR something if it's desired.

Or as an easier workaround, we could move all the calls for the app locale in the `Tag` class to a single method which can be overridden on a custom `Tag` class in user-land.

eg. This...

```php
$locale = $locale ?? app()->getLocale();
```

...would be elevated to a method `getTagLocale` defined on the Tag class.

```php
//public static $tagLocale = null;
public static $tagLocale = 'en'; // overridden in custom Tag class

public static function getTagLocale()
{
    return static::$tagLocale ?? app()->getLocale();
}
```

Then it would be used:

```php
public function scopeContaining(Builder $query, string $name, $locale = null): Builder
{
    $locale = $locale ?? static::getTagLocale();

    return [omitted];
}
```

This would at least allow me to force the en locale, regardless of the user locale. I don't love it though.

What are your thoughts?

Thanks for reading.
Lee

**Comment by mberneis** [comment]
I have exactly the same issue - Has there been a workaround?
